### PR TITLE
Single bit conditioning support for registerless circuits

### PR DESCRIPTION
**Pull request by TharrmashasthaPV**
### Summary

This PR enables single bit conditioning of gates and instructions in registerless circuits.

### Details and comments

This PR is an extension of #6018 to support single bit conditioning in registerless circuits. It is refactored over #6018 . Two tests are also added.

### CODE REVIEW

File: qiskit/circuit/instructionset.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def __init__(self):
         self.instructions = []
         self.qargs = []
         self.cargs = []
+        self.cregs = cregs
```
Comment (kdk) [inline qiskit/circuit/instructionset.py:32]: After talking with @1ucian0 , I think this is the right way of supporting `c_if(int, val)` right now. Can you add a comment here explaining why `cregs` is needed? (Also, maybe consider renaming as `circuit_cregs` or similar.)

**Review by enavarro51 — CHANGES_REQUESTED** [review]
It would be good to have a type/description here for `circuit_clbits`. Otherwise, looks good.

**Review by enavarro51 — APPROVED** [review]
LGTM. Thanks.

**Review by 1ucian0 — APPROVED** [review]
Thanks! once the single bit conditioning is fully supported, do you mind writing a release note with some details and examples?

**Pull request merged**